File: lib/swig/java/AbilityListComboBox.java
/*
 * Copyright (c) 2013-2014 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

package org.nc.PKMN;

import javax.swing.*;

public class AbilityListComboBox extends JComboBox<String>
{
    public AbilityListComboBox(int gen)
    {
        super();

        StringVector abilityVec = new StringVector();
        Lists.getAbilityList(abilityVec, gen);
        for(int i = 0; i < abilityVec.size(); i++)
        {
            addItem(abilityVec.get(i));
        }

        setEditable(false);
    }
}

File: lib/swig/cs/AbilityListComboBox.cs
/*
 * Copyright (c) 2014 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

using System.Windows.Forms;

namespace PKMN
{
    namespace Forms
    {
        public class AbilityListComboBox: ComboBox
        {
            public AbilityListComboBox(uint gen): base()
            {
                StringVector ability_vec = new StringVector();
                Lists.getAbilityList(ability_vec, gen);
                for(int i = 0; i < ability_vec.Count; i++)
                {
                    Items.Add(ability_vec[i]);
                }
            }
        }
    }
}
